### DISCUSSION TITLE
What is the difference between Location-based tracking and SLAM?

### DISCUSSION THREAD

**Discussion by Cloudinnovates** [AR-js-org/AR.js#375] [Ideas]
What is the difference between Location-based tracking and SLAM?
I am trying to wrap my head around it. I have used location-based tracking and it works well on dynamic LAT LON but I am trying to understand how does SLAM makes it different or how can we merge both technologies.

**Comment by nickw1** [comment]
Hi @Cloudinnovates, location-based AR involves using the device's GPS plus sensor readings (accelerometer, magnetic field sensor) to obtain the position and orientation of the device. Using these readings we can then position AR content in approximately its correct position within the world. Note however that due to inaccuracies in the GPS location (particularly) and the sensor readings (to a lesser extent) that the placement is not perfect.  However in many cases it's 'good enough'. The worst cases are when the augmented content is very close to the device as then, GPS inaccuracies have a proportionally bigger effect. Also, there is no plane detection so augmented content can appear to 'float' above the ground. See for example https://hikar.org/video/hikarweb.mp4

SLAM on the other hand involves building a map of the world using computer vision techniques to detect planes. Techniques typically involve identifying and tracking unique feature points (e.g. corners of a plane) from frame to frame. From the movement of feature points across multiple frame, the camera pose (position and orientation) within the world can be estimated, and planes can also be inferred from clusters of points.

It would be great to combine the two, as it would then mean we could detect planes in the real world and more realistically place the location-based content. However this is a difficult problem and currently there appears to be no open-source web solution to this. However I am, whenever I have time, investigating porting ORB-SLAM, one of the leading SLAM systems (see https://github.com/UZ-SLAMLab/ORB_SLAM3), to the web using Emscripten and WASM. I've done a little work on this so far though there's nothing demonstrable just yet.

**Reply by Cloudinnovates** [reply]
Thank You @nickw1 for the detailed insights.
I am too working on something similar to this https://hikar.org/video/hikarweb.mp4 but for sea. There are many uncertain variables that I have yet to test.
I am aware of ORB SLAM 3 but I will be more learning and understanding. Are you able to get any help from the  https://github.com/UZ-SLAMLab/ORB_SLAM3 team?
As you mention of no open source solutions. Do you happen to know of these solutions? Are you able to share with me? Thanks
Let me keep you posted here if I were to figure it out too.

**Reply by Cloudinnovates** [reply]
Do check out this for plane detection
https://immersive-web.github.io/webxr-samples/proposals/index.html

Hope this helps
